### PR TITLE
Add Publish Script

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What and why?

We want to make it as automatic as possible to release the package.

### How?

Add a github actions script to release the package.

### Review checklist

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Publish package on NPM
+on:
+  release:
+    types: [published]
```
Comment (juan-fernandez) [inline .github/workflows/publish.yml:4]: in https://github.com/DataDog/build-plugin/blob/master/.github/workflows/publish.yaml#L4 we have `[created]` but I think we this we avoid creating releases for drafts. This worked quite well for us at scope: https://github.com/DataDog/scope-javascript-agent/blob/master/.github/workflows/npmpublish.yml#L5

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '12.x'
+      - run: yarn config set npmAuthToken $YARN_NPM_AUTH_TOKEN
+        env:
+          YARN_NPM_AUTH_TOKEN: ${{ secrets.YARN_NPM_AUTH_TOKEN }}
+      - run: yarn
+      - run: yarn pack
+      - run: yarn publish
```
Comment (juan-fernandez) [inline .github/workflows/publish.yml:18]: if at some point we want to make beta releases we can with the "prerelease" option in github releases: https://github.com/DataDog/scope-javascript-agent/blob/master/.github/workflows/npmpublish.yml#L25

File: .github/workflows/publish.yml (hunk 1 above)
Comment (yoannmoinet) [reply .github/workflows/publish.yml:4]: Oh, that's interesting.

**Review by yoannmoinet — APPROVED** [review]
LGTM, thanks for doing it!

**Review by m-rousse — APPROVED** [review]
🎉

File: .github/workflows/publish.yml (hunk 1 above)
Comment (juan-fernandez) [reply .github/workflows/publish.yml:4]: sadly the docs around this are not great 😆 https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onevent_nametypes

**Pull request merged**